src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SuppressionTests/SuppressionEngineTests.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.ValidationSuppression.Tests
+{
+    public class SuppressionEngineTests
+    {
+        [Fact]
+        public void AddingASuppressionTwiceDoesntThrow()
+        {
+            var testEngine = SuppressionEngine.Create();
+            AddSuppression(testEngine);
+            AddSuppression(testEngine);
+
+            static void AddSuppression(SuppressionEngine testEngine) => testEngine.AddSuppression("PKG004", "A.B()", "ref/net6.0/mylib.dll", "lib/net6.0/mylib.dll");
+
+        }
+
+        [Fact]
+        public void SuppressionEngineCanParseInputSuppressionFile()
+        {
+            TestSuppressionEngine testEngine = TestSuppressionEngine.CreateTestSuppressionEngine();
+
+            // Parsed the right ammount of suppressions
+            Assert.Equal(10, testEngine.GetSuppressionCount());
+
+            // Test IsErrorSuppressed string overload.
+            Assert.True(testEngine.IsErrorSuppressed("CP0001", "T:A.B", "ref/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("CP0001", "T:A.C", "ref/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("CP0001", "T:A.B", "lib/netstandard2.0/tempValidation.dll", "lib/net6.0/tempValidation.dll"));
+            Assert.True(testEngine.IsErrorSuppressed("PKV004", ".netframework,Version=v4.8"));
+            Assert.False(testEngine.IsErrorSuppressed(string.Empty, string.Empty));
+            Assert.False(testEngine.IsErrorSuppressed("PKV004", ".netframework,Version=v4.8", "lib/net6.0/mylib.dll"));
+            Assert.False(testEngine.IsErrorSuppressed("PKV004", ".NETStandard,Version=v2.0"));
+
+            // Test IsErrorSuppressed Suppression overload.
+            Assert.True(testEngine.IsErrorSuppressed(new Suppression
+            {
+                DiagnosticId = "CP0001",
+                Target = "T:A.B",
+                Left = "ref/netstandard2.0/tempValidation.dll",
+                Right = "lib/net6.0/tempValidation.dll"
+            }));
+        }
+
+        [Fact]
+        public void SuppressionEngineDoesNotNeedFileToBeCreated()
+        {
+            SuppressionEngine engine = SuppressionEngine.CreateFromSuppressionFile("AFileThatDoesNotExist.xml");
```

Comment:
Shouldn't it ideally throw? This could potentially lead to bugs where the file path is wrong and we just don't know why the errors are being baselined?

Reply:
The idea here in terms of the experience, is that if you are running validation for the first time and want to baseline all of the errors, you will set the property that get's passed to here in your .csproj even when the file doesn't exist, which will just cause validation to create the file if it doesn't already exist. For example, one minimal project that will consume this should look like:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <Sdk Name="Microsoft.DotNet.PackageValidation" Version="1.0.0-dev" />

  <PropertyGroup>
    <TargetFrameworks>netstandard2.0;netstandard1.3;net6.0</TargetFrameworks>
    <!--Extra properties to be set for suppressions-->
    <BaselineAllPackageValidationErrors>true</BaselineAllPackageValidationErrors>
    <PackageValidationSuppressionBaselinePath>PackageValidationSuppressions.xml</PackageValidationSuppressionBaselinePath>
  </PropertyGroup>

</Project>
```

And in this example, even though PackageValidationSuppressions.xml file doesn't exist initially, after the first run of the validation the file will be generated for you.

Reply:
One thing to address before we merge the PR though, will be the name of those two properties: `BaselineAllPackageValidationErrors` `PackageValidationSuppressionBaselinePath`. Even though they are technically correct, the term 'baseline' is overloaded in package validation, as it already means checking against an old package, and in here I was more going for 'suppress all existing errors'.

Reply:
Yeah that makes sense; but wouldn't it make sense to throw if the file doesn't exist and we are note baselining errors? That would be the scenario where a user could potentially think that their suppressions are not working, so an error might help them figure out what the issue is, getting paths correctly on MSBuild can sometime cause trouble.

Reply:
would it make sense to divide this into 2 different properties, the input where we always check if the path exists and a another one to provide the path to the generated file.

Reply:
I think `PackageValidationSuppressionBaselinePath` should be the one that we use to generate when `BaselineAllPackageValidationErrors` is set to true. That way you don't need multiple settings.

Also, I think we should have a default, which would be a file at the project level with a convenient name.

Reply:
I added the default to the project directory, made it to throw if the file doesn't exist, but don't throw if the path is null or empty, because that means, the user didn't specify a suppressions file and the default path doesn't exist.

I also renamed the properties to: `CompatibilitySuppressionFilePath` and `GenerateCompatibilitySuppressionFile`